### PR TITLE
Orocos KDL

### PR DESCRIPTION
**Pull request by knusbaum**
New packages for Orocos KDL #8799 
The python packages fail to build because they depend on orocos-kdl. That is, when scanning for rdeps, it detects the shared library provided by orocos-kdl, but it's obviously not present yet.
If necessary, I can submit this as two PR's and wait on the python one until the other is merged.

The python packages each have their own template, since it's not an ordinary python build (not `build_style=python-module`) but is instead built with cmake.

### CODE REVIEW

File: srcpkgs/orocos-kdl/template [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+makedepends="eigen3.2"
+short_desc="Orocos Kinematics and Dynamics C++ library"
+maintainer="Kyle Nusbaum <knusbaum+void@sdf.org>"
+license="LGPL-2.1-only"
+homepage="http://www.orocos.org/kdl"
+distfiles="https://github.com/orocos/orocos_kinematics_dynamics/archive/v${version}.tar.gz"
+checksum=05b93e759923684dc07433ccae1e476d158d89b3c2be5079c20062406da7b4dd
+
+post_extract() {
+	mv ${XBPS_BUILDDIR}/orocos_kinematics_dynamics-${version}/orocos_kdl \
```
Comment (Johnnynator) [inline srcpkgs/orocos-kdl/template]: This sounds like you want to set ```
wrksrc="orocos_kinematics_dynamics-${version}"
build_wrksrc="orocos_kdl"
```

Comment (knusbaum) [reply srcpkgs/orocos-kdl/template]: Ahh, I wasn't aware of that variable. I'll change it in all the templates.

File: srcpkgs/python-orocos-kdl/template [hunk 2]
```diff
@@ -0,0 +1,17 @@
+# Template file for 'python-orocos-kdl'
+pkgname=python-orocos-kdl
+version=1.4.0
+revision=1
+build_style=cmake
+makedepends="python-sip-devel python-devel orocos-kdl eigen3.2"
+short_desc="Python 2 Bindings for Orocos Kinematics and Dynamics C++ library"
```
Comment (maxice8) [inline srcpkgs/python-orocos-kdl/template]: I'd rather not add a python2 library, is it really required ?

Comment (knusbaum) [reply srcpkgs/python-orocos-kdl/template]: I can eliminate it. When I asked on #voidlinux recently, the advice was to create both if possible.

**Comment by Chocimier** [comment]
> when scanning for rdeps, it detects the shared library provided by orocos-kdl, but it's obviously not present yet.

Not really. It is present, and installed because it is listed in depends, but to be found it needs to be added to common/shlibs file.

Please split off orocos-kdl-devel containing .so links, includes, pkgconfig and cmake files.

**Comment by knusbaum** [comment]
@Chocimier I will do that. I am curious why it's not found without the devel package, though.

File: srcpkgs/orocos-kdl/template [hunk 3]
```diff
@@ -0,0 +1,18 @@
+# Template file for 'orocos-kdl'
+pkgname=orocos-kdl
+version=1.4.0
+revision=1
+#wrksrc=orocos_kinematics_dynamics-${version}/orocos_kdl/
```
Comment (knusbaum) [inline srcpkgs/orocos-kdl/template]: forgot to delete this.

**Comment by Chocimier** [comment]
Devel package is unrelated to discovering library. What make it work is listing in common/shlibs that is a list of libraries to consider to link against between packages.

**Comment by knusbaum** [comment]
I see. Is that a manual process? I've never added such lines to common/shlibs before, and I don't see anything about it in the manual aside from mentioning the file's existence.

**Comment by Chocimier** [comment]
Yes, this is manual and this is the point, as not all shared objects are intended to be linked from other package like *private.so counterpart of public interface *.so, plugins for binary programs etc.

Yes, that could be documented better.

File: srcpkgs/python-orocos-kdl/template (hunk 2 above)
Comment (maxice8) [reply srcpkgs/python-orocos-kdl/template]: I wanna avoid adding new python2 stuff

File: srcpkgs/python3-orocos-kdl/template [hunk 4]
```diff
@@ -0,0 +1,15 @@
+# Template file for 'python3-orocos-kdl'
+pkgname=python3-orocos-kdl
+version=1.4.0
+revision=1
+wrksrc=orocos_kinematics_dynamics-${version}
+build_wrksrc=python_orocos_kdl
+build_style=cmake
+hostmakedepends="python3 python3-sip"
+makedepends="python3-sip-devel python3-devel orocos-kdl-devel eigen3.2"
+short_desc="Python 3 Bindings for Orocos Kinematics and Dynamics C++ library"
```
Comment (maxice8) [inline srcpkgs/python3-orocos-kdl/template]: Python3

Comment (knusbaum) [reply srcpkgs/python3-orocos-kdl/template]: Fixed

File: srcpkgs/python3-orocos-kdl/template [hunk 5]
```diff
@@ -0,0 +1,15 @@
+# Template file for 'python3-orocos-kdl'
+pkgname=python3-orocos-kdl
```
Comment (maxice8) [inline srcpkgs/python3-orocos-kdl/template]: needs to define PYTHON_VERSION -DPYTHON_VERSION=3

Comment (knusbaum) [reply srcpkgs/python3-orocos-kdl/template]: I'm assuming you mean in configure_args.

**Comment by maxice8** [comment]
On Wed, Feb 20, 2019 at 11:30:49AM -0800, Kyle Nusbaum wrote:
> knusbaum commented on this pull request.
> 
> 
> 
> > @@ -0,0 +1,15 @@
> +# Template file for 'python3-orocos-kdl'
> +pkgname=python3-orocos-kdl
> 
> I'm assuming you mean in configure_args. 
> 
> -- 
> You are receiving this because you commented.
> Reply to this email directly or view it on GitHub:
> https://github.com/void-linux/void-packages/pull/8965#discussion_r258641789

yes

File: srcpkgs/python3-orocos-kdl/template [hunk 6]
```diff
@@ -0,0 +1,18 @@
...
+makedepends="python3-sip-devel python3-devel orocos-kdl-devel eigen3.2"
+depends="python3 python3-sip"
+short_desc="Python3 Bindings for Orocos Kinematics and Dynamics C++ library"
+maintainer="Kyle Nusbaum <knusbaum+void@sdf.org>"
+license="LGPL-2.1-only"
+homepage="http://www.orocos.org/kdl"
+distfiles="https://github.com/orocos/orocos_kinematics_dynamics/archive/v${version}.tar.gz"
+checksum=05b93e759923684dc07433ccae1e476d158d89b3c2be5079c20062406da7b4dd
+
+configure_args="-DPYTHON_VERSION=3"
```
Comment (maxice8) [inline srcpkgs/python3-orocos-kdl/template]: add it directly below build_style=

Comment (knusbaum) [reply srcpkgs/python3-orocos-kdl/template]: ok. I tried xlint and it didn't complain.

**Pull request merged**